server/schedule/checker/rule_checker.go

```diff
@@ -258,6 +258,15 @@ func (c *RuleChecker) fixLooseMatchPeer(region *core.RegionInfo, fit *placement.
 		checkerCounter.WithLabelValues("rule_checker", "demote-voter-role").Inc()
 		return operator.CreateDemoteVoterOperator("fix-demote-voter", c.cluster, region, peer)
 	}
+	if core.IsVoter(peer) && core.IsWitness(peer) && !rf.Rule.IsWitness {
+		checkerCounter.WithLabelValues("rule_checker", "set-voter-non-witness").Inc()
+		return operator.CreateNonWitessVoterOperator("fix-demote-voter", c.cluster, region, peer)
+	}
+	if core.IsLearner(peer) && core.IsWitness(peer) && !rf.Rule.IsWitness {
```

Comment:
Will a learner also be a witness? I think it should be a kind of the voter.

Reply:
learner should be a witness, if it becomes witness after being voter, we need to clean the data applied just by in learner state, which is a waste.

Reply:
For `IsLearner` and `IsWitness` being confusing, do you have any better idea?

Reply:
What about adding a new `PeerRole`?

Reply:
@JmPotato There is a comparison discussion of `Scheme1: Witness as a flag (Selected)` and `Scheme2: Witness as a peer role` in the design document. Do you mean to support `Scheme2`? Would you please share your option in the design doc? Thanks!